### PR TITLE
😵‍💫 Add loading spinner to new document modal

### LINKED ISSUE #197: New Document Form: Submit indicator
Currently there is no indication that the form is being submitted when you press the "Create" button in the new document form. This is weird if the file is large enough to take a few seconds to upload

### LINKED ISSUE #213: Document creation can happen multiple times
Clicking the submit button quickly creates multiple copies of the same document.

### PR DESCRIPTION
**Pull request by phlmn**
closes #213
closes #197

### CODE REVIEW

**Comment by anuejn** [comment]
Would it be possible to have actual progress here?

**Comment by phlmn** [comment]
Sadly it's not that simple... :(

**Comment by phlmn** [comment]
@anuejn I would suggest creating a separate issue for a progress indicator

File: frontend/src/pages/new_document.tsx [hunk 1]
```diff
@@ -172,7 +179,9 @@ export function NewDocumentPage() {
               <SecondaryButton type="button" onClick={() => navigate(`/`)}>
                 Cancel
               </SecondaryButton>
-              <PrimaryButton type="submit">Create</PrimaryButton>
+              <PrimaryButton disabled={loading} type="submit">
+                {loading ? <LoadingSpinner /> : 'Create'}
+              </PrimaryButton>
```
Comment (pajowu) [inline frontend/src/pages/new_document.tsx:184]: Is it possible to have the spinner have the same size as the button?

Comment (phlmn) [reply frontend/src/pages/new_document.tsx]: Done

**Comment by phlmn** [comment]
> @anuejn I would suggest creating a separate issue for a progress indicator

#237

**Pull request merged**